src/modules/get_institution/app/get_institution_usecase.ts

```diff
@@ -0,0 +1,21 @@
+import { Institution } from "../../../core/repositories/database/models/Models";
+import { IInstitutionRepo } from "../../../core/repositories/interfaces/IInstitutionRepo";
+import { ConflictError, InvalidRequest, MissingParameter, UserNotAuthenticated } from '../../../core/helpers/errors/ModuleError';
+import { EntityError } from '../../../core/helpers/errors/EntityError';
+
+export class GetInstitutionUsecase {
+    public database_repo: IInstitutionRepo;
+    
+    constructor(database_repo: IInstitutionRepo) {
+        this.database_repo = database_repo;
+    }
+
+    public async execute(id: string) {
```

Comment:
U need to make all the logic of getting the authorization, check if the user is register... U can check in the other routes, how to do it!